components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt

```diff
@@ -24,6 +24,27 @@ import mozilla.components.service.glean.private.CounterMetricType
 import mozilla.components.service.glean.private.TimingDistributionMetricType
 import mozilla.components.service.glean.private.LabeledMetricType
 
+/**
+ * This component can emit metrics via Glean, but the consuming application
+ * needs to explicitly opt-in. It does so by calling this function to tell
+ * us what version of our metrics it wants to opt in to.
+ *
+ * Thus, it's very important that when you add new metrics-collection in this
+ * component, you increment the value of `LATEST_METRICS_VERSION` to indicate it.
+ */
+
+private const val LATEST_METRICS_VERSION = 1
+private const val METRICS_V1 = 1
+private var enabledMetricsVersion = 0
+
+@Suppress("TooGenericExceptionThrown")
+fun enableTelemetry(version: Int) {
+    if (version > LATEST_METRICS_VERSION) {
+        throw RuntimeException("mozilla.appservices.logins is on metrics version $LATEST_METRICS_VERSION, not $version")
+    }
+    enabledMetricsVersion = version
+}
```

Comment:
This may be extremely unidiomatic Kotlin, I'm happy to take suggestions for improvement.

Reply:
I think something like this would be more idiomatic/less error prone:

```kotlin
object LoginsTelemetry {
    enum class Version(val number: Int) {
        // Possibly better to use `null` for disabled and have a
        // separate `disable()` function...
        Disabled(0),
        V1(1),
    }
    @Synchronized
    fun enable(version: Version) {
        // check `version.value` against the known max version (see below)
        // update a private var, ...
    }
}
```

It would have the calling code looking like `LoginsTelemetry.enable(LoginsTelemetry.Version.V1)` or something (Prefixing this with logins avoids import being annoying for the user when we add PlacesTelemetry or whatever in the future). This would prevent the user from specifying a version higher than what we support.

We'd still probably want to check that the arg to enable has a version lower than our known max, to prevent the case where the calling code was compiled against a newer version of the logins package than it got at runtime, which can happen due to the compilation classpath being different from the runtime classpath, and having a newer version on the compilation classpath than runtime (aka gradle misconfiguration weirdness), but this would prevent accidents.